API_GUIDE.md

```diff
@@ -129,44 +178,79 @@ for epoch in range(1, num_epochs + 1):
+The only differences between the multithreading and multiprocessing code are:
+
+- Multiple devices are acquired in the same process with `xm.get_xla_supported_devices()`.
+- The model is wrapped in `dp.DataParallel` and passed both the training loop
+and dataloader.
+
+See the [full multithreading example](https://github.com/pytorch/xla/blob/master/test/test_train_mnist.py) for more on training a network on multiple XLA
+devices with multithreading.
+
+## XLA Tensor Deep Dive
 
-Check the [full example](https://github.com/pytorch/xla/blob/master/test/test_train_mnist.py) showing how to train MNIST on TPU using `torch_xla.distributed.data_parallel.DataParallel` (Python threading).
+Using XLA tensors and devices requires changing only a few lines of code. But
+even though XLA tensors act a lot like CPU and CUDA tensors their internals are
+different. This section describes what makes XLA tensors unique.
 
-## Discrepancies between PyTorch/XLA
+### XLA Tensors are Lazy
 
-PyTorch/XLA matches PyTorch eager mode user experience with a few exceptions imposed by the lazy tensor approach or implementation details.
-These differences don't affect performance, but might give "unexpected" results for normal PyTorch users.
+CPU and CUDA tensors launch operations immediately or `eagerly`. XLA tensors,
+on the other hand, are `lazy`. They record operations in a graph until the
+results are needed. Deferring execution like this lets XLA optimize it. A graph
+of multiple separate operations might be fused into a single optimized operation, for example.
 
-We list them in this section so that users are aware. They might get fixed in the future releases and updated here.
+Lazy execution is generally invisible to the caller. PyTorxh/XLA automatically
+constructs the graphs, sends them to XLA devices, and synchronizes when
+copying data between an XLA device and the CPU. Inserting a `barrier` when
+taking an optimizer step also synchronizes the CPU and the XLA device.
 
-1. Serialization of XLA tensors doesn't preserve view-relationship.
+### XLA Tensors and bFloat16
 
-   In normal PyTorch devices like CPU/CUDA, view-relationship is preserved when you save & load tensors sharing the same underlying storage.
+When Cloud TPUs are used as XLA devices then:
 
-   ``` Python
-   a = torch.rand(3, 3)
-   b = a[0]
-   c = a[0:2]
-   ```
-   That means loaded `b` and `c` still share the same storage. `c` is updated along with `b` and vice versa.
+- Floating point (float and double) XLA tensors use the same datatype
+- If the `XLA_USE_BF16` environment variable is set this datatype is `bfloat16`.
+If not set it's fp32.
 
-   In XLA case, `b` and `c` are separate tensors that one doesn't change with the other.
+XLA tensors on Cloud TPUs will report their PyTorch datatypes regardless of
```

Comment:
Remove Cloud. Just TPU. Whether on Cloud or on the ground, does not change the HW 😁

Reply:
Done.